### PR TITLE
[FLINK-7155] [metrics] Add Influxdb reporter

### PR DESCRIPTION
**Pull request by patricklucas**

### CODE REVIEW

File: flink-metrics/flink-metrics-influxdb/pom.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+		<dependency>
+			<groupId>io.dropwizard.metrics</groupId>
+			<artifactId>metrics-core</artifactId>
+			<version>${metrics.version}</version>
+		</dependency>
+
+		<dependency>
+			<groupId>io.dropwizard.metrics</groupId>
+			<artifactId>metrics-graphite</artifactId>
```
Comment (zentol) [inline flink-metrics/flink-metrics-influxdb/pom.xml]: why is this dependency needed?

File: flink-metrics/flink-metrics-influxdb/pom.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<artifactId>maven-assembly-plugin</artifactId>
+				<version>2.4</version>
+				<configuration>
+					<descriptorRefs>
+						<descriptorRef>jar-with-dependencies</descriptorRef>
```
Comment (zentol) [inline flink-metrics/flink-metrics-influxdb/pom.xml]: please use the maven-shade plugin instead.

File: flink-metrics/flink-metrics-influxdb/pom.xml (hunk 1 above)
Comment (patricklucas) [inline flink-metrics/flink-metrics-influxdb/pom.xml]: This was errant; I'll remove it.

File: flink-metrics/flink-metrics-influxdb/pom.xml (hunk 2 above)
Comment (patricklucas) [inline flink-metrics/flink-metrics-influxdb/pom.xml]: This is exactly how the graphite reporter does it, which is extremely similar to this one. I'm not sure exactly what you want me to change here.

Comment (zentol) [inline flink-metrics/flink-metrics-influxdb/pom.xml]: We intend to use the shade plugin for the graphite reporter as well, see https://issues.apache.org/jira/browse/FLINK-4173

**Comment by yew1eb** [comment]
Hi @patricklucas ,
Coding guidelines “No WIP pull requests.”  
Maybe you should see: http://flink.apache.org/contribute-code.html

BTW, If you do not have free time to finish it, I am willing to take over. :smile:

Best,
Hai Zhou

File: flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporter.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		final String user = config.getString(ARG_USER, null);
+		final String password = config.getString(ARG_PASSWORD, null);
+		final String db = config.getString(ARG_DB, "flink");
+		final String conversionRate = config.getString(ARG_CONVERSION_RATE, null);
+		final String conversionDuration = config.getString(ARG_CONVERSION_DURATION, null);
+
+		final metrics_influxdb.InfluxdbReporter.Builder builder =
+			metrics_influxdb.InfluxdbReporter.forRegistry(registry);
+
+		builder.protocol(new HttpInfluxdbProtocol(host, port, user, password, db));
```
Comment (gobozov) [inline flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporter.java]: We built same thing for our needs using same InfluxDb reporter. The disadvantage is that it does not support https, we modified it to support sslContext. I also don't see way to provide tags to reporter builder.

**Pull request closed without merging**